msal/changelog.txt

```diff
@@ -1,3 +1,17 @@
+2.4.0-preview
+=============
+- Cache fixes 
+   - Not remove account cache entities [PR](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1365)
+   - Fix issue with sharing in memory cahce for different instances of the TokenCache [PR](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1360)
+   - Clear in memory version of cache [MSAL issue #654](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/654)
+   - Fix NullPointException when adalv3 account is used because there is no client info in adalv3 [MSAL Issue #653](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/654)
+   - RemoveAdalUser api now removes users by accountID and by displayableID to cover migration scenario from adalv3 to adalv4 to msalv2. [MSAL Issue #652](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/652)
+   - Fix for internal TokenCache.Clear api so ADAL persistent cache is also cleared [MSAL Issue #651](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/651)
+   - Optimize cache lookups with native iOS by modifying field for persistence on iOS. [PR](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1350)
+- Support [extended lifetime tokens](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1331)
+- Remove Chrome dependency for system browser on Android devices. See https://aka.ms/msal-net-system-browsers for more information. [MSAL issue #664](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/664)
```

Comment:
>Remove Chrome dependency for system browser [](start = 2, length = 43)

:)

Reply:
:face_with_rolling_eyes:

---
In reply to: [231366239](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1389#discussion_r231366239) [](ancestors = 231366239)

Reply:
I would propose:
2.4.0-preview
=============
Improvements and fixes to the token cache
- The serialized token cache can now be shared by different applications, therefore providing SSO if the same user signs-in in both applications
  - See [PR](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1365) and [MSAL Issue #653](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/654)
- On .NET Core, the Token cache was shared by all instances of applications in memory. This is now fixed (See MSAL.NET issue #656 and [PR](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1360))
- Fixes consistency issues for advanced token cache migration scenarios from ADAL v3.x to ADA v4.x to MSAL v2.x 
  - [MSAL Issue #652](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/652)
  - [MSAL Issue #651](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/651)
- Cache lookups were optimized. Work done in conjunction with ADAL.iOS and MSAL.iOS native) [PR](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1350)

More browsers are now supported on Xamarin.Android when you choose to use system web browsers.
- Removed chrome dependency for system browser on Android devices. See https://aka.ms/msal-net-system-browsers for more information. [MSAL issue #664](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/664)

Azure AD B2C improvement
- Add support for b2clogin.com for b2c authorities [MSAL issue #669](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/669) [#632](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/632)

Reply:
this is great...thank you. we needed your PM magic. I  did a copy/paste

---
In reply to: [231409353](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1389#discussion_r231409353) [](ancestors = 231409353)